cloudca/resource_cloudca_volume.go

```diff
@@ -0,0 +1,204 @@
+package cloudca
+
+import (
+	"fmt"
+	"github.com/cloud-ca/go-cloudca"
+	"github.com/cloud-ca/go-cloudca/api"
+	"github.com/cloud-ca/go-cloudca/services/cloudca"
+	"github.com/hashicorp/terraform/helper/schema"
+	"strings"
+)
+
+func resourceCloudcaVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudcaVolumeCreate,
+		Read:   resourceCloudcaVolumeRead,
+		Update: resourceCloudcaVolumeUpdate,
+		Delete: resourceCloudcaVolumeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"service_code": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A cloudca service code",
+			},
+			"environment_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of environment where port forwarding rule should be created",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the volume to be created",
+			},
+			"zone_name": &schema.Schema{
```

Comment:
Ok. I was under the impression that we wanted to move away from supporting name and ID because it could get confusing and made things complicated in the provider.

Looks like the AWS provider uses availability zone names only, but the CloudStack one does accept either zone name or ID.